libmscore/volta.cpp

```diff
@@ -51,7 +51,7 @@ static const ElementStyle voltaStyle {
 //   VoltaSegment
 //---------------------------------------------------------
 
-VoltaSegment::VoltaSegment(Spanner* sp, Score* s) : TextLineBaseSegment(sp, s, ElementFlag::MOVABLE | ElementFlag::ON_STAFF)
+VoltaSegment::VoltaSegment(Spanner* sp, Score* s) : TextLineBaseSegment(sp, s, ElementFlag::MOVABLE | ElementFlag::ON_STAFF | ElementFlag::SYSTEM)
```

Comment:
Isn't this violating the current meaning of `SYSTEM` flag a bit? As far as I understand the system flag is assigned to elements that are drawn once for the whole system and do not belong to any particular staff. This doesn't seem to be true for volta segments, though there probably should be a way to mark that the element should use system-wide style settings.

Reply:
Voltas are special.  They are already being added with the system flag set, but only on the Volta itself, not the VoltaSegments, and this caused problems once I started scaling lines - some aspects of the volta would scale, others wouldn't, depending on whether they were handled in Volta or VoltaSegment.  So I needed to make these consistent.

A little background:

When added to the top staff, voltas *are* treated as regular system elements - they appear on the top visible staff in the score and on all staves of the score.  So the system flag is absolutely appropriate.  However, there were enough requests to be able to also add voltas on lower staves that at some point, this capability was added.  Voltas added to staves *other* than the top staff get added and have the system flag set, but then they are special-cased in excepts.cpp so the volta does not get exported to parts.

Probably it makes more sense to only set the system flag when actually being added to the top staff.  And actually, fix it so it gets set when added to the top *visible* staff - right now that's a bug (see https://musescore.org/en/node/277265).

Anyhow, the way we treat voltas with respect to the SYSTEM flag for voltas is a bit of a hack indeed, but it's not my hack; I'm just cleaning it up a little by making sure it also gets set for the volta segments.

FWIW, I would say, we could change the volta add code to only set the system flag when adding to the top visible staff (and force the track to 0 even if there are invisible staves above).  But then, I worry about cases where the volta is added then staves are rearranged, and you end up with a volta on the top staff that is *not* set as SYSTEM.  But in any case, changing the basic behavior of voltas seemed outside the scope of this PR, so I didn't change it.

Reply:
Thanks for your explanation! I agree, it is indeed better to keep that system flag consistent with the volta itself.